### ISSUE TITLE
Editting frame won't save

### ISSUE THREAD

**Issue by MathieuDR** [gochrono/chrono#2]
### Expected behaviour:
When editing a frame, closing the editor, closes the editor command and saves the frame.

### Actual behaviour:
When editing a frame, closing the editor closes the 'editor command' but does not save the frame.

### version
Version: dev
Commit: none
Built: 2018-12-26T19:36:26Z

### Notes:
- Editor is Visual studio code for me (Correct JSON program)
- Deleting a frame also doesn't happen unless I add -f to it. (I do press 'y' for confirmation)

**Comment by JordanKnott** [comment]
Hello! Thanks for using Chrono and for filing a bug report.

> Deleting a frame also doesn't happen unless I add -f to it. (I do press 'y' for confirmation)

Commit https://github.com/gochrono/chrono/commit/bf7b0371b7a42b70200d33977350e9315b3ea81c should fix this issue.

> Editor is Visual studio code for me (Correct JSON program)

Could you run the `edit` command with the `verbose` then send the output?

**Comment by MathieuDR** [comment]
Sure

```
λ chrono edit 0 --verbose
INFO 2019/02/01 00:37:12 using config dir: {path}\AppData\Roaming\chrono
INFO 2019/02/01 00:37:12 using storage type msgpack
INFO 2019/02/01 00:37:12 reading frames from {path}\AppData\Roaming\chrono\frames.msgpack
INFO 2019/02/01 00:37:37 saving frames [&{[{23825c28-affe-4eaf-a52a-e1c517e23c9d german 2019-01-30 00:14:49.1892847 +0100 CET 2019-01-30 00:32:06.9807736 +0100 CET 2019-01-30 00:14:49.1892847 +0100 CET [learning germany tag] [Phrases Basics 2 Adjectives]} {31954932-926d-451c-858f-3534b025eb67 League 2019-01-30 00:39:37.7591056 +0100 CET 2019-01-30 00:51:26.8943486 +0100 CET 2019-01-30 00:39:37.7591056 +0100 CET [game fun] [taliyah]} {bb8d01fb-6b61-46cc-8a92-d6e0ec4c5db6 tools 2019-01-30 21:42:13.9623531 +0100 CET 2019-01-30 22:19:08.1152129 +0100 CET 2019-01-30 21:42:13.9623531 +0100 CET [dev] [already busy for 15 mins.]} {8ab68540-3f85-41d8-accc-a464d2346012 League 2019-01-30 22:19:19.6680346 +0100 CET 2019-01-30 22:36:36.7610375 +0100 CET 2019-01-30 22:19:19.6680346 +0100 CET [game fun] []} {82af9acc-7992-4626-aaee-062b2875114e german 2019-01-30 23:31:01.044487 +0100 CET 2019-01-31 00:01:59.1540162 +0100 CET 2019-01-30 23:31:01.044487 +0100 CET [learning germany tag] [Plurals to 1 Phrases to 4 Accusative Case to 2]} {44316d6a-dc7e-4276-ae82-80f920843a10 League 2019-01-31 19:24:49.3107001 +0100 CET 2019-01-31 19:57:38.4008086 +0100 CET 2019-01-31 19:24:49.3107001 +0100 CET [fun game ranked] [Xerath lost, lux was better at roaming, i sppose. stole baron]} {a35af609-92ba-4c15-af2e-87969f436e55 AOE 2019-01-31 20:20:00 +0100 CET 2019-01-31 21:10:00 +0100 CET 2019-01-31 21:27:39.5923431 +0100 CET [fun game] [levi en cas. KOTH - bots (moderate)]} {f8eae6cf-427e-40b2-b27b-8b44c0430cc5 German 2019-01-31 21:32:45.2099662 +0100 CET 2019-01-31 21:51:11.670057 +0100 CET 2019-01-31 21:32:45.2099662 +0100 CET [germany learning tag] [Accusative form Case to 2 Intro to 4 Questions to 1 Not to 1 Practice]} {0292a6c8-38a9-4bb5-9ba0-25cae3f8388f League 2019-01-31 22:13:00 +0100 CET 2019-01-31 22:47:37.4705603 +0100 CET 2019-01-31 22:47:29.9860734 +0100 CET [fun game ranked] [Xerath]} {50068092-6243-4cc8-b556-af6906e14516 IdentityService 2019-01-31 22:47:00 +0100 CET 2019-01-31 23:28:29.6269477 +0100 CET 2019-01-31 23:28:23.7421728 +0100 CET [poluza dev work] [Pluralsight learning]}]}] to {path}\AppData\Roaming\chrono\frames.msgpack
INFO 2019/02/01 00:37:37 serialzied frames: [129 166 70 114 97 109 101 115 154 135 164 85 85 73 68 217 36 50 51 56 50 53 99 50 56 45 97 102 102 101 45 52 101 97 102 45 97 53 50 97 45 101 49 99 53 49 55 101 50 51 99 57 100 167 80 114 111 106 101 99 116 166 103 101 114 109 97 110 169 83 116 97 114 116 101 100 65 116 215 255 45 33 5 112 92 80 222 105 167 69 110 100 101 100 65 116 215 255 233 213 171 128 92 80 226 118 169 85 112 100 97 116 101 100 65 116 215 255 45 33 5 112 92 80 222 105 164 84 97 103 115 147 168 108 101 97 114 110 105 110 103 167 103 101 114 109 97 110 121 164 108 105 110 97 165 78 111 116 101 115 147 167 80 104 114 97 115 101 115 168 66 97 115 105 99 115 32 50 170 65 100 106 101 99 116 105 118 101 115 135 164 85 85 73 68 217 36 51 49 57 53 52 57 51 50 45 57 50 54 100 45 52 53 49 99 45 56 53 56 102 45 51 53 51 52 98 48 50 53 101 98 54 55 167 80 114 111 106 101 99 116 166 76 101 97 103 117 101 169 83 116 97 114 116 101 100 65 116 215 255 180 252 33 0 92 80 228 57 167 69 110 100 101 100 65 116 215 255 213 58 180 224 92 80 230 254 169 85 112 100 97 116 101 100 65 116 215 255 180 252 33 0 92 80 228 57 164 84 97 103 115 146 164 103 97 109 101 163 102 117 110 165 78 111 116 101 115 145 167 116 97 108 105 121 97 104 135 164 85 85 73 68 217 36 98 98 56 100 48 49 102 98 45 54 98 54 49 45 52 54 99 99 45 56 97 57 50 45 100 54 101 48 101 99 52 99 53 100 98 54 167 80 114 111 106 101 99 116 165 116 111 111 108 115 169 83 116 97 114 116 101 100 65 116 215 255 229 113 95 48 92 82 12 37 167 69 110 100 101 100 65 116 215 255 27 120 9 144 92 82 20 204 169 85 112 100 97 116 101 100 65 116 215 255 229 113 95 48 92 82 12 37 164 84 97 103 115 145 163 100 101 118 165 78 111 116 101 115 145 185 97 108 114 101 97 100 121 32 98 117 115 121 32 102 111 114 32 49 53 32 109 105 110 115 46 135 164 85 85 73 68 217 36 56 97 98 54 56 53 52 48 45 51 102 56 53 45 52 49 100 56 45 97 99 99 99 45 97 52 54 52 100 50 51 52 54 48 49 50 167 80 114 111 106 101 99 116 166 76 101 97 103 117 101 169 83 116 97 114 116 101 100 65 116 215 255 159 69 152 160 92 82 20 215 167 69 110 100 101 100 65 116 215 255 181 114 10 240 92 82 24 228 169 85 112 100 97 116 101 100 65 116 215 255 159 69 152 160 92 82 20 215 164 84 97 103 115 146 164 103 97 109 101 163 102 117 110 165 78 111 116 101 115 144 135 164 85 85 73 68 217 36 56 50 97 102 57 97 99 99 45 55 57 57 50 45 52 54 50 54 45 97 97 101 101 45 48 54 50 98 50 56 55 53 49 49 52 101 167 80 114 111 106 101 99 116 166 103 101 114 109 97 110 169 83 116 97 114 116 101 100 65 116 215 255 10 155 69 96 92 82 37 165 167 69 110 100 101 100 65 116 215 255 36 184 103 32 92 82 44 231 169 85 112 100 97 116 101 100 65 116 215 255 10 155 69 96 92 82 37 165 164 84 97 103 115 147 168 108 101 97 114 110 105 110 103 167 103 101 114 109 97 110 121 164 108 105 110 97 165 78 111 116 101 115 147 172 80 108 117 114 97 108 115 32 116 111 32 49 172 80 104 114 97 115 101 115 32 116 111 32 52 180 65 99 99 117 115 97 116 105 118 101 32 67 97 115 101 32 116 111 32 50 135 164 85 85 73 68 217 36 52 52 51 49 54 100 54 97 45 100 99 55 101 45 52 50 55 54 45 97 101 56 50 45 56 48 102 57 50 48 56 52 51 97 49 48 167 80 114 111 106 101 99 116 166 76 101 97 103 117 101 169 83 116 97 114 116 101 100 65 116 215 255 74 19 161 16 92 83 61 113 167 69 110 100 101 100 65 116 215 255 95 143 106 96 92 83 69 34 169 85 112 100 97 116 101 100 65 116 215 255 74 19 161 16 92 83 61 113 164 84 97 103 115 147 163 102 117 110 164 103 97 109 101 166 114 97 110 107 101 100 165 78 111 116 101 115 146 166 88 101 114 97 116 104 217 54 108 111 115 116 44 32 108 117 120 32 119 97 115 32 98 101 116 116 101 114 32 97 116 32 114 111 97 109 105 110 103 44 32 105 32 115 112 112 111 115 101 46 32 115 116 111 108 101 32 98 97 114 111 110 135 164 85 85 73 68 217 36 97 51 53 97 102 54 48 57 45 57 50 98 97 45 52 99 49 53 45 97 102 50 101 45 56 55 57 54 57 102 52 51 54 101 53 53 167 80 114 111 106 101 99 116 163 65 79 69 169 83 116 97 114 116 101 100 65 116 214 255 92 83 74 96 167 69 110 100 101 100 65 116 214 255 92 83 86 24 169 85 112 100 97 116 101 100 65 116 215 255 141 57 192 240 92 83 90 59 164 84 97 103 115 146 163 102 117 110 164 103 97 109 101 165 78 111 116 101 115 145 217 35 108 101 118 105 32 101 110 32 99 97 115 46 32 75 79 84 72 32 45 32 98 111 116 115 32 40 109 111 100 101 114 97 116 101 41 135 164 85 85 73 68 217 36 102 56 101 97 101 54 99 102 45 52 50 55 101 45 52 48 98 50 45 98 50 55 98 45 56 98 52 52 99 48 52 51 48 99 99 53 167 80 114 111 106 101 99 116 166 71 101 114 109 97 110 169 83 116 97 114 116 101 100 65 116 215 255 50 15 81 224 92 83 91 109 167 69 110 100 101 100 65 116 215 255 159 193 8 160 92 83 95 191 169 85 112 100 97 116 101 100 65 116 215 255 50 15 81 224 92 83 91 109 164 84 97 103 115 147 167 103 101 114 109 97 110 121 168 108 101 97 114 110 105 110 103 164 108 105 110 97 165 78 111 116 101 115 149 185 65 99 99 117 115 97 116 105 118 101 32 102 111 114 109 32 67 97 115 101 32 116 111 32 50 170 73 110 116 114 111 32 116 111 32 52 174 81 117 101 115 116 105 111 110 115 32 116 111 32 49 168 78 111 116 32 116 111 32 49 168 80 114 97 99 116 105 99 101 135 164 85 85 73 68 217 36 48 50 57 50 97 54 99 56 45 51 56 97 57 45 52 98 98 53 45 57 98 97 48 45 50 53 99 97 101 51 102 56 51 56 56 102 167 80 114 111 106 101 99 116 166 76 101 97 103 117 101 169 83 116 97 114 116 101 100 65 116 214 255 92 83 100 220 167 69 110 100 101 100 65 116 215 255 112 48 184 176 92 83 108 249 169 85 112 100 97 116 101 100 65 116 215 255 235 25 36 224 92 83 108 241 164 84 97 103 115 147 163 102 117 110 164 103 97 109 101 166 114 97 110 107 101 100 165 78 111 116 101 115 145 166 88 101 114 97 116 104 135 164 85 85 73 68 217 36 53 48 48 54 56 48 57 50 45 54 50 52 51 45 52 99 99 56 45 98 53 53 54 45 97 102 54 57 48 54 101 49 52 53 49 54 167 80 114 111 106 101 99 116 175 73 100 101 110 116 105 116 121 83 101 114 118 105 99 101 169 83 116 97 114 116 101 100 65 116 214 255 92 83 108 212 167 69 110 100 101 100 65 116 215 255 149 121 217 208 92 83 118 141 169 85 112 100 97 116 101 100 65 116 215 255 176 242 162 0 92 83 118 135 164 84 97 103 115 147 166 112 111 108 117 122 97 163 100 101 118 164 119 111 114 107 165 78 111 116 101 115 145 180 80 108 117 114 97 108 115 105 103 104 116 32 108 101 97 114 110 105 110 103]
Edited frame for project German, from 00:14:49 to 00:32:06 (0h 17m 17s)
```

it seems that it notices the changes, but doesn't actually save them.
Since the next time i look at my frames or the frame in particular, it didn't change.

Am I a version behind?

**Comment by JordanKnott** [comment]
Seems I never actually implemented the function that updates frames. Oops!
I pushed a commit (https://github.com/gochrono/chrono/commit/e4900674d37d55ac56c6626eb0ff14aa0e20c613) that should fix the `edit` command not working.

>Am I a version behind?

I've pushed several commits today so it's possible.

> If you want to talk about some other things (small things that's maybe not worth a full bug report unless you want me too,) you can add me on discord: ySomic#6703

I sent you a friend invite on Discord, I'd be happy to hear any suggestions or issues you've found!
